### PR TITLE
Use Lwt_io for output and other minor changes

### PR DESCRIPTION
**Pull request by mseri**
This makes the executable more verbose but makes it easier to spot issues

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

### CODE REVIEW

File: src/vncproxy.ml [hunk 1]
```diff
@@ -62,42 +62,58 @@ let exn_to_string = function
...
+           else Lwt.return_none)
+        (fun e ->
+          (* Lwt_io.eprintlf "warning: %s" (Printexc.to_string e) >>= fun () -> *)
+          Lwt.return_none)
+    ) cs >>= fun all -> 
+  match all with
+  | [ exactly_one ] -> Lwt.return exactly_one
   | [] ->
-    Printf.fprintf stderr "The VM is exposing no VNC consoles.\n%!";
-    fail (Failure (Printf.sprintf "The VM is exposing no VNC consoles"))
```
Comment (gaborigloi) [inline src/vncproxy.ml]: It might be useful to extract this `eprintlf` + `fail_with` pattern into a helper function in a future PR.

Comment (mseri) [reply src/vncproxy.ml]: We could probably simplify the code further in many places, but I did not want to spend too much time (it took already long enough to pinpoint what was failing)

**Comment by mseri** [comment]
It's a cli executable, I thought Lwt_low was overkill :P but feel free to update it

**Comment by mseri** [comment]
This fixes https://github.com/xapi-project/vncproxy/issues/7

**Pull request merged**